static/js/stream_list.js

```diff
@@ -108,13 +108,17 @@ exports.build_stream_list = function () {
 
     _.each(stream_groups.pinned_streams, add_sidebar_li);
 
-    if (stream_groups.pinned_streams.length > 0) {
+    var any_pinned_streams = stream_groups.pinned_streams.length > 0;
+    var any_normal_streams = stream_groups.normal_streams.length > 0;
+    var any_dormant_streams = stream_groups.dormant_streams.length > 0;
```

Comment:
In JS, any number that isn't 0 is evaluated as a truthy value. So I would suggest changing `var any_pinned_streams = stream_groups.pinned_streams.length > 0` to `var any_pinned_streams = stream_groups.pinned_streams.length`, and similarly for the other 2 lines.

Reply:
In my opinion, the reduction of two characters is not worth it as it diminishes readability.

Reply:
Agreed.  When doing interactive development, using truthyness is great, but for production code, you usually want to spell the check out to make the intent clearer to future contributors to that code.